explanations/dispatchers.ipynb

```diff
@@ -0,0 +1,825 @@
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "res = spacemapped(\n",
+    "    **sc_space.dense_vars,\n",
+    "    **sc_space.sparse_vars,\n",
+    "    consumption=1,\n",
+    ")\n",
+    "res"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(3, 4)"
+      ]
+     },
+     "execution_count": 23,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "res.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lets try to get this result via looping over the grids and calling `utility` directly"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Array([[-0.5, -0.5, -0.5, -0.5],\n",
+       "       [ 0. ,  0. ,  0. ,  0. ],\n",
+       "       [ 1. ,  2. ,  3. ,  4. ]], dtype=float32)"
+      ]
+     },
+     "execution_count": 24,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "_res = jnp.empty((3, 4))\n",
+    "\n",
+    "# loop over valid combinations of sparse variables (first axis)\n",
+    "for i, (lagged_retirement, retirement) in enumerate(\n",
+    "    zip(\n",
+    "        sc_space.sparse_vars[\"lagged_retirement\"],\n",
+    "        sc_space.sparse_vars[\"retirement\"],\n",
+    "        strict=False,\n",
+    "    ),\n",
+    "):\n",
+    "\n",
+    "    # loop over product of dense variables\n",
+    "    for j, wealth in enumerate(sc_space.dense_vars[\"wealth\"]):\n",
+    "\n",
+    "        u = utility(\n",
+    "            wealth=wealth,\n",
+    "            retirement=retirement,\n",
+    "            lagged_retirement=lagged_retirement,\n",
+    "            consumption=1,\n",
+    "        )\n",
+    "        _res = _res.at[i, j].set(u)  # JAX arrays are immutable\n",
+    "\n",
+    "_res"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If `put_dense_first` would be False, the order of the loops would have been switched,\n",
+    "leading to an output shape of (4, 3)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "### Explanation of Results\n",
+    "\n",
+    "The outputs align with the utility function: The rows represent sparse combinations of _lagged_retirement_ and _retirement_, while the columns represent values of _wealth_. For column $j$, the value of _wealth_ corresponds to $j$.\n",
+    "\n",
+    "Consider the first row, corresponding to _lagged_retirement_ being 0 and _retirement_ also being 0. In this scenario, the agent is working, incurring a cost of -0.5. Since the agent is not retired, there is no bequest utility, resulting in a utility of $log(1) - 0.5 = -0.5$ for all _wealth_ values.\n",
```

Comment:
Yes, you are completely right; this is nonsense.

Because it works very nicely for the example, I want a term in the utility function that is 0 if  _lagged_retirement_ is 0 and equal to _wealth_ if _lagged_retirement_ is 1.

My current idea would be to call it _retirement utility_, or do you have a different suggestion? The idea would be that once you are retired for at least one period, you get utility corresponding to your wealth level.

Reply:
`retirement_habit` ? Seems closest to a habit formation thing. 

Not worth spending much time on it, though, just bequests are too misleading imo.